File: typescript/pangram/pangram.ts
export default class Pangram {
    constructor(private readonly s: string) {
    }

    isPangram() {
        const alphabet = new Set<string>('abcdefghijklmnopqrstuvwxyz'.split(''))
        const set = new Set<string>(this.s.split('').filter(c => alphabet.has(c)))
        return set.size === alphabet.size
    }
}

File: haskell/pangram/src/Pangram.hs
module Pangram
  ( isPangram
  ) where

import Data.Char (toLower, isLetter)
import Data.HashSet (fromList)

isPangram :: String -> Bool
isPangram text
  | null characters = False
  | length characters == length alphabet = True
  | otherwise = False
  where
    characters = fromList [toLower c | c <- text, isLetter c]
    alphabet = ['a' .. 'z']